chore: Update and cleanup project

friederbluemle opened the pull request:
## Description

A bunch of project cleanups and updates. The changes are in individual commits for easier review. Please let me know if there are any questions with individual patches. See explanations below for more information:

### Clean up lock files

A project should have **either** a `yarn.lock` file **or** a `package-lock.json` file. The existence of `yarn.lock` in the root folder suggests usage of Yarn which results in the following warning:

    package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.

The React Native sample project in the `example` folder _should_ use Yarn (`yarn.lock` *already* added in 85f17d01f7824ab29ff4e698a5c2101e50e9d134) - I deleted the redundant `package-lock.json` file here.

### Remove extra Gradle wrapper

There was a misplaced Gradle wrapper sitting in the `android/` folder. The executable bit of `gradlew` was not even set. Removed the unnecessary scripts together with the wrapper jar file.

### Update .gitignore

Consolidated the .gitignore files and removed some duplications

### Update example project

Minor UI updates and improvements, as well as committed the automated updates to the Xcode project file by CocoaPods (`pod install`) to avoid a dirty working directory after running this necessary set-up step.

### Simplify Podfile and add **Podfile.lock**

There were some redundant/extraneous declarations/statements in `Podfile`. Also adds the missing `Podfile.lock`.

### Update refresh-example script

Added the `pod install` step in the `ios` folder of the example project. Running `./refresh-example.sh` should now result in **zero changes** (if the dependencies are up-to-date). Also reordered a couple of steps to remove the temporary directory earlier (pod install requires `Podfile` that was saved in the temp folder).

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |
| Windows |    ✅ *   |

\* Due to lack of devices/OS I was not able to test the Windows part.

Tested on Android and iOS devices and emulator/simulator. Opened and successfully ran the project both in Android Studio 3.3.2 and Xcode 10.2

mikehardy commented:
Nice!! Thank you!

Can I ask you to use the refresh-example script and only commit the things necessary in the example to implement your changes, not the things that vary from the clean result of the refresh? That helps me keep track of what is really changing, and what I really need to change as new react-native versions come out. For example, I do not commit the XCode perturbations because it seems to me they are pointless and always happen so have little value, but knowing when the link command has altered behavior on a clean project is very valuable. Just doing some git reset --soft HEAD~1 or similar work and cleaning those changes can help you keep your directory clean for commits

I really appreciate the chore work though, that's what helps this project remain supportable in the face of new users that aren't familiar with react-native processes...

mikehardy commented:
I like this in theory but will close it if feedback is not handled, sorry

friederbluemle commented:
@mikehardy - So sorry for the long delay! I've had a lot going on recently (got married last week! ;))

Anyways, I just rebased the branch and updated this PR. I reverted the changes to the Xcode `project.pbxproj` file. Please check again. Thanks!

mikehardy commented:
Hey! Best excuse ever, congratulations :-). Okay, I'll check this out again, thanks

mikehardy commented on example/package.json:
```diff
@@ -4,20 +4,17 @@
   "devDependencies": {
     "@babel/core": "7.4.0",
     "@babel/runtime": "7.4.2",
-    "@react-native-community/eslint-config": "0.0.3",
```
The eslint items are things we are in the process of implementing per react-native-community standards - can you leave those in? Also, just because everything moves quickly - if you're in there RN0.59.8 is out now, but it's just some minor version bumps - easy to include: https://github.com/react-native-community/rn-diff-purge/compare/release/0.59.5..release/0.59.8

That said, could you use the refresh-example script to make the changes, to insure that it works and tracks the example refreshes correctly? RN0.60 is coming shortly and I want to make sure the example refresh is just a single command line away...

friederbluemle commented:
I noticed the eslint dependencies were already removed on `master` due to the changes in `85f17d0` - So that comment is already obsolete ;)

mikehardy commented:
Yeah, I updated the example yesterday as I was working on other things :-), best to just leave the example alone - I obliterate it semi-frequently with the refresh script and I have an interest in keeping that working to minimize support load, so we can leave all of it out here really. Let me look through the other stuff

mikehardy commented:
Well - this started going in a direction that I'm not comfortable with at the moment. pod install as a requirement is not on my list things I want

I know RN0.60 is going there but the pod install requirement itself for project startup was a matter of some debate, and we (react-native-community / react native) tried to avoid it. It was not possible but for RN0.59 (still current) I do not want to add it yet as it is a startup barrier. I want to do it only when strictly required by the RN dependency (which would be whenever I refresh the example to be RN0.60)

Can you just drop all the example changes? Even the ones that are for imports and formatting make it skew farther from the template, and will make it more difficult to track template changes (because App.js will change with RN0.60 as well)

friederbluemle commented:
> Well - this started going in a direction that I'm not comfortable with at the moment.

Hmm not sure what you mean. I did not add CocoaPods in this PR... It's not really part of this PR to remove the Podfile. Is this something you want/not want?

I also just updated the main PR description. Could you take a look and let me know if that clarifies things a bit?

mikehardy commented:
Sorry if I was unclear, I mean with the yarn dep in the example, and the pod install + Podfile.lock. We do have a Podfile in the example but that was also support-load related. We got so many pod issues I dropped an example in there I could point to, but we are not emphasizing it yet so I had not (and don't want to) start directing people to pod things and make it a requirement yet, lest I become the leading edge of support for react-native's switch to pods by default :-). I would like to be in the middle of the adopter curve there.

So I'd like to leave the example alone completely for now

friederbluemle commented:
Rebased and resolved conflicts (again). It would be great to get this merged soon. Thanks!

friederbluemle commented:
I removed the commits that update the project file and add the Podfile.lock. Just as before, running `pod install` will leave a dirty working directly and untracked files.

mikehardy commented:
re: conflicts, indeed, we're a beehive here right now - maintaining PRs is not a lot of fun. Thanks for working with me. Also thanks for going with me on the example bit. I'm okay with locally dirty files in the example for the moment for the reasons mentioned. We'll have that area cleaner shortly after RN0.60 releases (and if you want to collaborate there since your Pod kung-fu seems stronger than mine, it will be very welcome)

friederbluemle commented:
Sounds good, thanks @mikehardy :+1:
